### PR TITLE
chore: add bun to ci

### PR DESCRIPTION
**Pull request by juliusmarminge**
Follows up https://github.com/trpc/trpc/pull/4716

### CODE REVIEW

**Comment by juliusmarminge** [comment]
cc @colinhacks - I've yet to try out Bun so welcome for any pointers to make the example better

**Comment by colinhacks** [comment]
Looks great! Perfect minimal implementation w/ `Bun.serve`

File: examples/bun/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+- [Bun](https://bun.sh)
+- Vanilla TRPCClient in Deno
```
Comment (kamilogorek) [inline examples/bun/README.md:2]:
```suggestion
- Vanilla tRPC Client in Bun
```

File: examples/bun/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+[Install Bun](https://bun.sh/docs/installation) then run:
+
+```sh
+bun run src/server.ts
```
Comment (kamilogorek) [inline examples/bun/README.md:7]:
```suggestion
bun run src/index.ts
```

Comment (kamilogorek) [reply examples/bun/README.md:7]: Or maybe rely on `bun run` being able to call `package.json` scripts directly with `bun start:server` and `bun start:client`?

File: examples/bun/src/client.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  const proxy = createTRPCProxyClient<AppRouter>({
+    links: [loggerLink(), httpBatchLink({ url })],
+  });
+
+  await delay(100);
+
+  // parallel queries
+  await Promise.all([
+    //
```
Comment (kamilogorek) [inline examples/bun/src/client.ts:21]:
```suggestion
```

**Review by kamilogorek — CHANGES_REQUESTED** [review]

File: examples/bun/README.md (hunk 2 above)
Comment (juliusmarminge) [reply examples/bun/README.md]: yep that's better

**Review by kamilogorek — APPROVED** [review]

**Pull request merged**